src/types/AuthContext.ts

```diff
@@ -0,0 +1,32 @@
+import { ajv } from '../ajv';
+import { userSchema } from './User';
+import type { JSONSchemaType } from 'ajv';
+import type { User } from './User';
+
+interface AuthContext {
```

Comment:
I mention this in the commit message, but to highlight it here: by having a more deterministic definition of AuthContext we can pass an auth context around to other places (such as loader functions).  The sooner we "tighten" the type to signal that both user and role ARE present, the better.

Reply:
Double-plus good!